include/devicetree/pwms.h

```diff
@@ -0,0 +1,383 @@
+/**
+ * @file
+ * @brief PWMs Devicetree macro public API header file.
+ */
+
+/*
+ * Copyright (c) 2020, Linaro Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DEVICETREE_PWMS_H_
+#define ZEPHYR_INCLUDE_DEVICETREE_PWMS_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup devicetree-pwms Devicetree PWMs API
+ * @ingroup devicetree
+ * @{
+ */
+
+/**
+ * @brief Get pwm controller "name" (label property) at an index
+ *
+ * It's an error if the pwm controller referenced by the phandle
+ * in property "pwms" at index "idx" has no label property.
+ *
+ * Example devicetree fragment:
+ *
+ *     pwm1: pwm-controller@... {
+ *             label = "PWM_1";
+ *     };
+ *
+ *     pwm2: pwm-controller@... {
+ *             label = "PWM_2";
+ *     };
+ *
+ *     n: node {
+ *             pwms = <&pwm1 1 PWM_POLARITY_NORMAL>,
+ *                    <&pwm2 3 PWM_POLARITY_INVERTED>;
+ *     };
+ *
+ * Example usage:
+ *
+ *     DT_PWMS_LABEL_BY_IDX(DT_NODELABEL(n), 1) // "PWM_2"
+ *
+ * @param node_id node identifier
+ * @param idx logical index into the property
+ * @return the label property for the referenced node at index idx
+ * @see DT_PROP_BY_PHANDLE_IDX()
+ */
+#define DT_PWMS_LABEL_BY_IDX(node_id, idx) \
+	DT_PROP_BY_PHANDLE_IDX(node_id, pwms, idx, label)
+
+/**
+ * @brief Get pwm controller "name" (label property) by name
+ *
+ * It's an error if the pwm controller referenced by the phandle
+ * in property "pwms" at index "idx" has no label property.
```

Comment:
copy/paste problem here; "idx" is not a macro parameter

Reply:
fixed